rust/gui-client/src-tauri/src/client/gui/system_tray.rs

```diff
@@ -84,62 +77,108 @@ impl Tray {
             update_ready: state.release.is_some(),
         };
 
-        self.handle.set_tooltip(TOOLTIP)?;
-        self.handle.set_menu(build_app_state(state))?;
+        let menu = state.into_menu();
+        let menu_clone = menu.clone();
+        let app = self.app.clone();
+        let handle = self.handle.clone();
+
+        if Some(&menu) == self.last_menu_set.as_ref() {
+            tracing::debug!("Skipping redundant menu update");
+        } else {
+            self.app
+                .run_on_main_thread(move || {
+                    if let Err(error) = update(handle, &app, &menu) {
+                        tracing::error!(?error, "Error while updating tray icon");
+                        firezone_telemetry::capture_anyhow(&error);
```

Comment:
Should we enable the `tracing` layer so we don't need to double these up all the time?